src/shared/settingsConfiguration.ts

```diff
@@ -56,4 +59,38 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
             return false
         }
     }
+
+    public readDevSetting<T>(key: AwsDevSetting): string
```

Comment:
Nit: does this description still require a `T`/ can you do something like `T = string`?

Reply:
The generic type should be constrained to the primitive types (since that's what we're actually checking), then we can use a little bit of TypeScript magic to infer the generic type. But this is overkill :)

Reply:
> then we can use a little bit of TypeScript magic to infer the generic type.

@JadenSimon that would be really useful, if it will allow us to throw an error if the type is not correct. This would avoid weird problems if e.g. `readDevSetting<number>('foo')` is called but the user config has stored a string in the "foo" setting.

Reply:
Yes we can get rid of the type parameter as long as we associate the 'real' type with each key (e.g. with a dictionary). For non-primitive types (and their keys), they could be associated with a 'validator' function to achieve true type-safety.

Reply:
I threw this together (untested!) but shows the type inference:
[Playground link](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABANwIYBsYBNVQKYByIAtgEZ4BOAghRagJ4AUcpAVgFyKpj0CUnYEuQoBtALqIA3gFgAUIgWIYwRIwCENOvQB0MAM6aGzNr0QAfM4hatteuMTyNkiALwA+RGsZR6ABzxwKs4uIYgA5IJklGG8sVJyiomIUAAWFHAA7ohgeFkAorRwFIwRcFBcSKi0DFYqkcJ6MQmKAL7NChR4UCAUSNZybbJyoJCwCCgY2Lh4AMr2eACSYPgUwKgQjtac3HyckojA6KgA5pykcHDoeNwANFa+Y2AYAPycelAUMGDHiC3x8oplKo1NZzJZGAB5Nh4aDaADWeHoemMrF42iu31SiA8ACZEAAyfHJPwBFTWbSHE6eUJhc6Xa5gMIEontJIo7RwB7wJ7oVyhcBYPDAL54LBg4n+QJWNgcrkIDB8lzhd6fb4xOIyAFJZJpTLZXKIArpYqlcrcLjVei1bJCSiNXiswaJTrdXrS1gDORyAD03p1eEQvk+xBgsGQeD0FoDrBA70DRw2KUugookeARQleAAtF9gJQ8JA8HI8AAPXxFcoQBBxmZ5AAqdYWBAA4gB9ADSeQAmjNXP9EiIwqgMnptILw+hOZQKRcAIxhMScMJhG6swfD0fjvCT-wUGdwHELzgABlXWsQ65HY7wE6ne-TcAAzEeJpgcPgiFFqJazwOh1etx3acHwAFhfNA32mOYHCWFY1g2M9BjkHx-EQGYulgb49AAWVQXw+xQ0k0PrRsWw7bsZmQklECoEcABEb3QqBMJ+JUEXoKUmJYnC8Ko1D20ROs4DrEkAB460QUt8DALBI1ovQGOQLivmODwlWUrDcN8EQ6wkKSC1k1QtgqPhXA8Rhc0oRAACVTEsHJwwoRBnhsxBOA044eO03SvVkUtywocovjg9YAxmKhsLrPBiF8I58DyMBkBgdIwAcZYADUqhgVBSCuSNNQHTo7B6DY3g+FTF37bULzhMrVWOSqVRU8wbS-Fq6SubhHU9IZZF9RA9BgDEgrTDA9ADUgrUFNYQHQKBhnAaBuUQTpUCwRSPNE1kJP0mS5PoxiMJUu5WV7XbDI6hk5DcRh2M4Os7kG4bXjQ-hEAE+ghJE-xxI8cERBmPSS2kwyRG8CgQADSwBSFEUsF4CQXJh4UcjFAQb0oUwCsUKswDjCCpigDN1JIps207HsRHYiQvnebgNilAAxRbHkQVlEhc2sGzJ8jKeprhIxRbYeFMdwTLZ89Ek4IWTNFjxseqwEVG8EkpVBNRQkIzjSbIimZipxExCx9nFcUVJ0iyHJ8kKE1CMQEM9GIXAIBSUUmkl6qnVNjouh6Po2BNhQkPPXG41x4VWJQPQq0FbQMiKOE9F8ULtGOLoAGEEAjnpcG5RgHV6xIPitBWkhdP3X0JopGHDmBjlTrpbsRWJHUQCBnZSVRKAoY2PYUfqsDgRAMhSaZHMQV3OkDpRlbUJ6CygXvvYUc29W76evaD1lWX6mOA2TrFuAwJF9CUSNwByq5kiHyynNSXBwhmJlT7wABHEAYAghfr-CD5IaZSaOpT7lmCqycubpkZwwFiZHqchQ7lCyEqVa61DrMRUiUDc15by7n3POAu8DEAlj7MgjaR1vgYIAhjIC94LiHjuH-PA+DqzlCtEg64KClJkOOBQzcVC7z7mfEwvG5QABexD2GkLQeQ-8vDsHAQuGBXgQA)